### ISSUE TITLE
All objc builds are broken on kokoro

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#14291]
Started failing last night, 100% failure rate since than:
First failing builds:
https://kokoro2.corp.google.com/job/grpc/job/macos/job/master/job/grpc_basictests_dbg/1390/
https://kokoro2.corp.google.com/job/grpc/job/macos/job/master/job/grpc_basictests_opt/1386/

https://sponge.corp.google.com/invocation?id=8c1f3ea6-03d9-4f69-8f81-ab9a9dc1a616&searchFor=

Not sure if it's infrastructure problem or a bad PR merged.

**Comment by jtattermusch** [comment]
@dgquintas  @muxi

**Comment by jtattermusch** [comment]
Please check if it could be this PR https://github.com/grpc/grpc/pull/14285.

**Comment by dgquintas** [comment]
#14285 can't be the cause: it's a protected method that's only used in the same file it's defined.

**Comment by matt-kwong** [comment]
It looks like it might be a clang error:
```
clang: error: -fobjc-weak is not supported on the current deployment target
2018-02-01 21:21:48.232 xcodebuild[5195:705601] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Logs/Test/91A00000-24FD-4CDD-807E-3617D521C3A1/Session-AllTests-2018-02-01_212145-EcKjcD.log" UserInfo={NSLocalizedDescription=Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Logs/Test/91A00000-24FD-4CDD-807E-3617D521C3A1/Session-AllTests-2018-02-01_212145-EcKjcD.log}
2018-02-01 21:21:48.234 xcodebuild[5195:705762] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7fb901adef20>
Testing failed:
	-fobjc-weak is not supported on the current deployment target
```
Let me check if the MacOS image has been updated.

**Comment by muxi** [comment]
FYI I created [a verbose run](https://sponge.corp.google.com/invocation?tab=Build+Log&id=bf774a96-e383-4956-93f8-b3295216a2cd&searchFor=) earlier today to troubleshoot and it turns out for some reason the tag `-fobjc-weak` was added into the build command; it is not there on my local build.

Checked with apolcyn that Ruby was not modified last night. tools/internal_ci did not seem to be changed too.

**Comment by matt-kwong** [comment]
It looks like the image was updated last night, but rolled back this morning. Waiting on a build to see if this fixes the problem.

**Comment by matt-kwong** [comment]
I tried testing on v1.9.x and ObjC passes: https://kokoro2.corp.google.com/job/grpc/job/macos/job/pull_request/job/grpc_basictests_dbg/2559/. I don't see MacOS tests failing in the PRs merged last night, so perhaps a combination of them causes the issue?

**Comment by muxi** [comment]
I created #14302 and #14303 to pinpoint which was the cause. #14302 reverts master to the state before 6:18PM Feb 1; #14303 reverts changes in `tools/internal_ci` to the state in v1.9.x. We'll see which one(s) fail.

**Comment by muxi** [comment]
#14302 still failing and #14303 passed. Looks like something is breaking in CI script...

**Comment by jtattermusch** [comment]
Current situation: many of the objc examples are now building fine, but for the `objc-tests` target, I see this compilation error:
``
write-file /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-RxLibrary.build/gRPC-RxLibrary-own-target-headers.hmap
Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/RxLibrary/GRXConcurrentWriteable.m:70:12: error: block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior [-Werror,-Wimplicit-retain-self]
      if (!_alreadyFinished) {
           ^
           self->
Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/RxLibrary/GRXConcurrentWriteable.m:71:9: error: block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior [-Werror,-Wimplicit-retain-self]
        _alreadyFinished = YES;
        ^
        self->
2 errors generated.
...
The following build commands failed:
	CompileC /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-RxLibrary.build/Objects-normal/i386/GRXConcurrentWriteable.o /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/RxLibrary/GRXConcurrentWriteable.m normal i386 objective-c com.apple.compilers.llvm.clang.1_0.compiler
```

**Comment by jtattermusch** [comment]
https://github.com/grpc/grpc/blob/b1f5d59b4fb567cdcc6786122d7bfcbd698f9af0/src/objective-c/RxLibrary/GRXConcurrentWriteable.m#L70

**Comment by jtattermusch** [comment]
Hopefully fixed by https://github.com/grpc/grpc/pull/14314

**Comment by matt-kwong** [comment]
Fixed by #14314 or #14324. Last ObjC test suite passed for both dbg and opt in master.
